pkg/operator/workload/workload_openshiftapiserver_v311_00_sync_test.go

```diff
@@ -219,3 +237,155 @@ func TestPreconditionFulfilled(t *testing.T) {
 		})
 	}
 }
+
+func TestCapabilities(t *testing.T) {
+	testCases := []struct {
+		name                    string
+		knownCapabilities       []configv1.ClusterVersionCapability
+		enabledCapabilities     []configv1.ClusterVersionCapability
+		expectedPerGroupOptions []openshiftcontrolplanev1.PerGroupOptions
+	}{
+		{
+			name:                    "Build and DC capabilities enabled implicitly",
+			knownCapabilities:       []configv1.ClusterVersionCapability{},
+			enabledCapabilities:     []configv1.ClusterVersionCapability{},
+			expectedPerGroupOptions: []openshiftcontrolplanev1.PerGroupOptions{},
+		},
+		{
+			name: "Build and DC capabilities enabled explicitly",
+			knownCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			enabledCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			expectedPerGroupOptions: []openshiftcontrolplanev1.PerGroupOptions{},
+		},
+		{
+			name: "Build capability disabled",
+			knownCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			enabledCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			expectedPerGroupOptions: []openshiftcontrolplanev1.PerGroupOptions{
+				{Name: openshiftcontrolplanev1.OpenShiftBuildAPIserver, DisabledVersions: []string{"v1"}},
+			},
+		},
+		{
+			name: "DC capability disabled",
+			knownCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			enabledCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+			},
+			expectedPerGroupOptions: []openshiftcontrolplanev1.PerGroupOptions{
+				{Name: openshiftcontrolplanev1.OpenShiftAppsAPIserver, DisabledVersions: []string{"v1"}},
+			},
+		},
+		{
+			name: "Build and DC capabilities disabled",
+			knownCapabilities: []configv1.ClusterVersionCapability{
+				configv1.ClusterVersionCapabilityBuild,
+				configv1.ClusterVersionCapabilityDeploymentConfig,
+			},
+			enabledCapabilities: []configv1.ClusterVersionCapability{},
+			expectedPerGroupOptions: []openshiftcontrolplanev1.PerGroupOptions{
+				{Name: openshiftcontrolplanev1.OpenShiftBuildAPIserver, DisabledVersions: []string{"v1"}},
+				{Name: openshiftcontrolplanev1.OpenShiftAppsAPIserver, DisabledVersions: []string{"v1"}},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			kubeClient := fake.NewSimpleClientset(
```

Comment:
please rename it to `fakeKubeClient`

Reply:
TestOperatorConfigProgressingCondition uses kubeClient well. Do you wanna rename it on other test(s) as well?

Reply:
sure, if this is not a problem for you that would be fine :)

Reply:
Done